### PR TITLE
Request invite page

### PR DESCRIPTION
**Pull request by EfrenRey**
@albertoarias @paulino-corujo @arnaud-elasticbox

### CODE REVIEW

**Comment by allyourbotsarebelongtous** [comment]
Deployed test instance is available at http://ek-106.internal.elastickube.com

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/ek_pull_request/150/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Deployed test instance is available at http://ek-106.internal.elastickube.com

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/ek_pull_request/151/ for more details.

**Comment by albertoarias** [comment]
@EfrenRey This looks good!

I found the following issues:
- Disable the button after Request Invite is pressed. It takes a little bit and it is disorienting as it seems nothing is happening
- If the invite text is empty we should not show in the email:
image
-  If I am not logged in and I try to send invite from the email I am presented with the login screen but after I login the flow is abandoned. i.e. it forgets that I am trying to send an invite.

@mattnickles you can do the flow yourself in the PR instance using:
https://ek-106.external.elastickube.com

**Comment by osanjuan** [comment]
@EfrenRey can you take care of alberto's comments? thanks!

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/ek_pull_request/155/ for more details.

**Comment by arnaud-elasticbox** [comment]
Jenkins, test this please.

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/ek_pull_request/159/ for more details.

**Comment by albertoarias** [comment]
@EfrenRey can you make sure this passes the test?

**Comment by albertoarias** [comment]
@EfrenRey also look at my comment

**Comment by EfrenRey** [comment]
@albertoarias The first 2 points of comment are already addressed... working on the last one.

**Comment by EfrenRey** [comment]
Jenkins, test this please

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/ek_pull_request/166/ for more details.

**Comment by EfrenRey** [comment]
@albertoarias Now the redirect after login seems work.
@manolakis Could you review the changes in auth-run.js and ek-login-controller.js Thanks

**Comment by allyourbotsarebelongtous** [comment]
Deployed test instance is available at http://ek-106.internal.elastickube.com

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/ek_pull_request/168/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Instance i-i78ee4 couldnt be deleted. It requires manual deletion

**Comment by allyourbotsarebelongtous** [comment]
The following instances are being terminated: https://elasticbox.com/services/instances/i-i78ee4, https://elasticbox.com/services/instances/i-uzq3qc

File: src/ui/app/config/auth-run.js [hunk 1]
```diff
@@ -26,6 +26,8 @@ function checkRouteAccess($rootScope, auth, initialization, instancesNavigationA
                     if (auth.isLoggedIn()) {
                         instancesNavigationActionCreator.instances();
                     } else {
+                        sessionStorage.setItem("toState", JSON.stringify(toState))
```
Comment (manolakis) [inline src/ui/app/config/auth-run.js]: @EfrenRey create just one entry into the session store storing an object with both items, and use a semantical key in order to identify what are we storing. `toState` and `toParams` are not saying anything to me if I don't have the context.

File: src/ui/app/login/ek-login/ek-login.controller.js [hunk 2]
```diff
@@ -24,12 +24,23 @@ class LoginController {
...
     }
 
     submit() {
         return this._principalActionCreator.login(this.user)
             .then(() => this._initialization.initializeLoggedInUser())
-            .then(() => this._instancesNavigationActionCreator.instances())
+            .then(() => {
+                if (sessionStorage.hasOwnProperty("toState")) {
+                    var _toState = JSON.parse(sessionStorage.getItem("toState"));
+                    var _toParams = JSON.parse(sessionStorage.getItem("toParams"));
```
Comment (manolakis) [inline src/ui/app/login/ek-login/ek-login.controller.js]: use `const` instead `var`

File: src/ui/app/core/module.js [hunk 3]
```diff
@@ -74,6 +74,7 @@ angular
     ])
 
     .constant('storage', localStorage)
+    .constant('sessionStorage', sessionStorage)
```
Comment (manolakis) [inline src/ui/app/core/module.js]: Don't use the sessionStorage. We have a sessionStore item and everything we want to store in session should be managed through that service

File: src/ui/app/login/ek-login/ek-login.controller.js [hunk 4]
```diff
@@ -15,7 +15,7 @@ limitations under the License.
 */
 
 class LoginController {
-    constructor($log, $scope, initialization, instancesNavigationActionCreator, loginNavigationActionCreator, principalActionCreator) {
+    constructor($log, $scope, initialization, instancesNavigationActionCreator, loginNavigationActionCreator, principalActionCreator, routerHelper, sessionStorage) {
```
Comment (manolakis) [inline src/ui/app/login/ek-login/ek-login.controller.js]: Use the session service

File: src/ui/app/admin/ek-admin-users/ek-admin-users.controller.js [hunk 5]
```diff
@@ -77,15 +78,18 @@ class AdminUsersController {
         usersStore.addChangeListener(onUsersChange);
 
         $scope.$on('$destroy', () => usersStore.removeChangeListener(onUsersChange));
+        if (this._$stateParams.invite) {
```
Comment (manolakis) [inline src/ui/app/admin/ek-admin-users/ek-admin-users.controller.js]: Maintain the `$scope.$on('$destroy', ...` as the last block of the constructor.

Move this block of code over the definition of the `tableOptions`.

File: src/ui/app/admin/ek-invite-users/ek-invite-users.controller.js [hunk 6]
```diff
@@ -25,7 +25,7 @@ class InviteUsersController {
 
         this.namespaces = namespacesStore.getAll();
         this.selectedNamespaces = [];
-        this.emails = [];
+        this.emails = this.emails || [];
```
Comment (manolakis) [inline src/ui/app/admin/ek-invite-users/ek-invite-users.controller.js]: Inside the constructor `this.emails` can't be defined previously, no?

File: src/ui/app/config/auth-run.js [hunk 7]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-function checkRouteAccess($rootScope, auth, initialization, instancesNavigationActionCreator, loginNavigationActionCreator, routerHelper) {
+function checkRouteAccess($rootScope, auth, initialization, instancesNavigationActionCreator, loginNavigationActionCreator, routerHelper, sessionStorage) {
```
Comment (manolakis) [inline src/ui/app/config/auth-run.js]: Use the session service instead the sessionStorage

File: src/ui/app/login/ek-request-invite/ek-request-invite.less [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+        font-weight: 300;
+    }
+
+    &__legend {
+        font-size: 14px;
+        margin-top: 0;
+    }
+
+    &__form {
+
```
Comment (manolakis) [inline src/ui/app/login/ek-request-invite/ek-request-invite.less]: remove this blank line

File: src/ui/app/login/ek-request-invite/ek-request-invite.less [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+        }
+
+        &__submit-btn {
+            width: 220px;
+            height: 45px;
+            margin: 0 0 45px 0;
+            font-weight: bold;
+            color: @white !important;
+            border-radius: 2px;
+            &.md-raised:not([disabled]) {
```
Comment (manolakis) [inline src/ui/app/login/ek-request-invite/ek-request-invite.less]: add a blank line before and after inner blocks

File: src/ui/app/login/ek-login/ek-login.controller.js [hunk 10]
```diff
@@ -24,12 +24,23 @@ class LoginController {
...
         return this._principalActionCreator.login(this.user)
             .then(() => this._initialization.initializeLoggedInUser())
-            .then(() => this._instancesNavigationActionCreator.instances())
+            .then(() => {
+                if (sessionStorage.hasOwnProperty("toState")) {
+                    var _toState = JSON.parse(sessionStorage.getItem("toState"));
+                    var _toParams = JSON.parse(sessionStorage.getItem("toParams"));
+                    sessionStorage.removeItem("toState");
+                    sessionStorage.removeItem("toParams");
+                    this._routerHelper.changeToState(_toState.name, _toParams);
```
Comment (manolakis) [inline src/ui/app/login/ek-login/ek-login.controller.js]: Minor thing: I think we should use some ActionCreator. @paulino-corujo ?

File: src/ui/app/login/ek-request-invite/ek-request-invite.controller.js [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+
+        if (this._$stateParams.account) {
+            this.account = this._$stateParams.account;
+            this.name = this._$stateParams.name;
+        }
+    }
+
+    submit() {
+        this.submitting = true;
+        return this._principalActionCreator.requestInvite({ email: this.account || this.email, name: this.name })
```
Comment (manolakis) [inline src/ui/app/login/ek-request-invite/ek-request-invite.controller.js]: Add blank line before the return.

File: src/ui/app/admin/ek-admin-users/ek-admin-users.controller.js [hunk 12]
```diff
@@ -77,15 +78,18 @@ class AdminUsersController {
...
 
-    inviteUsers() {
+    inviteUsers(emails) {
         const settings = this._settingsStore.getSettings();
 
         if (_.isUndefined(settings.mail)) {
             return this._adminNavigationActionCreator.warnOutboundEmailDisabled();
         }
-        return this._adminNavigationActionCreator.inviteUsers();
+        return this._adminNavigationActionCreator.inviteUsers(emails);
```
Comment (manolakis) [inline src/ui/app/admin/ek-admin-users/ek-admin-users.controller.js]: Add a blank line before the return

File: src/ui/app/admin/ek-admin-users/ek-admin-users.controller.js (hunk 5 above)
Comment (manolakis) [inline src/ui/app/admin/ek-admin-users/ek-admin-users.controller.js]: Add blank line before the if block

**Pull request merged**